### ISSUE TITLE
Latest nuget does not handle link css in head

### ISSUE THREAD

**Issue by jjxtra** [lukeschafer/HtmlCleanser#13]
I would expect .css links in the head to be inlined, but this tool just strips them out without pulling their contents first and inlining them.

**Comment by lukeschafer** [comment]
Correct. There are numerous issues with pulling them in, such as trustworthiness of sources (XSS attacks etc), relative vs fully-qualified, etc.

If you really want to inline external stylesheets, you can add your own rule to do it, but we would need a very compelling reason to include it by default.

To write your own rule (this uses a very archaic form of composition sorry), implement `IHtmlCleanserRule` then when initialising the cleanser, pass an instance into the constructor. You will also need to pass the type of `DocumentShouldNotReferenceResources` as an ignored type when you call cleanse, as it will remove the script tag prior to your rule running. Looking at it, it's actually cumbersome as it won't then inline it, so y ou would need to mess around with the style inliner somehow. I will change some code so it's easier


Remember this lib can be used for far more than just cleansing outgoing razor-templated emails. For example, when it was first created it was used while brokering HTML emails between users.

**Comment by jjxtra** [comment]
I'll give it a try, thanks for the fast response!

**Comment by lukeschafer** [comment]
No problem, easier to deal with as soon as I see it :)

I've pushed a PR to better support you and I will merge it once the build finishes. https://github.com/lukeschafer/HtmlCleanser/pull/14

This exposes the underlying perform method. All you need to do is send it exactly the set of rules you want, in the order you want them. So if you create a rule to inline external sources, you can have that happen first, then the style inlining

**Comment by jjxtra** [comment]
I'm using this in https://github.com/DigitalRuby/MailDemon. Hoping to move away from premail :)

Is there a rule built in that will do the pulling of the css contents and inline them? Or do I have to write that myself?

**Comment by lukeschafer** [comment]
You would have to write it yourself. I'm not in a position to do it today,
but I could conceivably do something soon, or if you want you can run your
implementation by me
